highlight/highlight
# Allow users to sort traces

## Summary

Allows users to sort traces by clicking the column headers of the traces table.

https://www.loom.com/share/cd09c436b33548d0bdf6add86439a98e

## How did you test this change?

* Click table headers to trigger sorting
  * Ensure results are sorted correctly
  * Ensure that scrolling loads the correct results
* Ensure scrolling backward on logs loads the correct results
* Ensure all results load as expected for logs and traces

## Are there any deployment considerations?

We will want to monitor performance when using sorting. My testing was in development without a lot of data, but it could have a more drastic negative impact on performance in production.

## Does this work require review from our design team?

Have been discussing the UX of table headers w/ @julian-highlight in https://linear.app/highlight/issue/HIG-4525/improvements-to-table-headers which will include some additional changes to table headers.
